cdn-server/cdn/src/index.ts

```diff
@@ -0,0 +1,100 @@
+import { JWTVerifyResult, jwtVerify } from 'jose';
+import { Context, Env, Hono, Next, Schema } from 'hono';
+
+export interface BlobStorage {
+  getObject(context: Context, key: string): Promise<ReadableStream>;
+}
+
+export class BlobNotFoundError extends Error {
+  constructor(message: string, cause?: Error) {
+    super(message, cause);
+    Object.setPrototypeOf(this, BlobNotFoundError.prototype);
+  }
+}
+
+interface CdnOptions {
+  authJwtSecret: string | ((c: Context) => string);
+  blobStorage: BlobStorage;
+}
+
+declare module 'hono' {
+  interface ContextVariableMap {
+    authenticatedOrganizationId: string;
+    authenticatedFederatedGraphId: string;
+  }
+}
+
+const jwtMiddleware = (secret: string | ((c: Context) => string)) => {
```

Comment:
Please add at least a few tests for both middlewares.

Reply:
I've added tests for both, PTAL